mono/mini/mini-amd64.c

```diff
@@ -2715,10 +2715,7 @@ mono_arch_start_dyn_call (MonoDynCallInfo *info, gpointer **args, guint8 *ret, g
 	pindex = 0;
 
 	/* Stored after the stack arguments */
-	nullable_buffer = (guint8*)&(p->regs [PARAM_REGS + (dinfo->nstack_args * sizeof (mgreg_t))]);
```

Comment:
was the old array indexer here a bug? just curious, since it looks like a bug fix

Reply:
Correct, this was a bug, so for a use case currently just hit on Windows this caused stack corruption and crash, https://github.com/mono/mono/pull/11786/commits/6d0af3c44dd322700ce6f3240bd80f4d9a0f2b7d